src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -367,10 +367,17 @@ class ResourceService(
         throw t
     }
 
-  def addSubjectToPolicy(policyIdentity: FullyQualifiedPolicyId, subject: WorkbenchSubject): Future[Unit] =
-    directoryDAO.addGroupMember(policyIdentity, subject).unsafeToFuture().map(_ => ()) andThen {
-      case Success(_) => fireGroupUpdateNotification(policyIdentity)
+  def addSubjectToPolicy(policyIdentity: FullyQualifiedPolicyId, subject: WorkbenchSubject): Future[Unit] = {
+    subject match {
+      case subject: FullyQualifiedPolicyId if policyIdentity.resource.resourceTypeName.equals(ManagedGroupService.managedGroupTypeName) =>
+        throw new WorkbenchExceptionWithErrorReport(ErrorReport(s"Access policies cannot be added to managed groups."))
+      case _ =>    {
+        directoryDAO.addGroupMember(policyIdentity, subject).unsafeToFuture().map(_ => ()) andThen {
```

Comment:
Is using `andThen` safe here?  I think I would rather this thing only return when and if the directoryDAO update completed successfully and the `fireGroupUpdateNotification` completed successfully.

Reply:
let's leave the `andThen` eradication for a later PR... I want to make sure this has good enough performance first